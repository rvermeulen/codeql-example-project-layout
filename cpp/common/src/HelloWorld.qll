bindingset[s]
predicate isAHelloWorld(string s) { s.matches("Hello World%") }
