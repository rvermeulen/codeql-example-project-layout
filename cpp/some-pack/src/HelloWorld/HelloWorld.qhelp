<!DOCTYPE qhelp PUBLIC
  "-//Semmle//qhelp//EN"
  "qhelp.dtd">
<qhelp>
<overview>
<p>Finds incorrect usage of the string 'Hello World'</p>

</overview>
<recommendation>

<p>Switch to using the prefix 'Hello World'</p>

</recommendation>
<example>

<p>The following example shows both a correct and incorrect way of using a 'Hello World' string.</p>

<sample src="HelloWorldExample.cpp" />

</example>
<references>

<li>
Foo:
<a href="https://">Bar</a>.
</li>

</references>
</qhelp>
