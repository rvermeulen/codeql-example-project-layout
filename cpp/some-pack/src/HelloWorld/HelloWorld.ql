/**
 * @kind problem
 * @precision high
 * @problem.severity error
 * @tags hello-world
 */

import cpp
import HelloWorld // Import support library from common

from StringLiteral s
where
  exists(s.getValue().regexpFind("([hH]ello)|([wW]orld)", _, _)) and
  not isAHelloWorld(s.getValue())
select s
