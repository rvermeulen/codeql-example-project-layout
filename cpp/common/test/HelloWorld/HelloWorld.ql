import cpp
import HelloWorld

from StringLiteral s
where isAHelloWorld(s.getValue())
select s
