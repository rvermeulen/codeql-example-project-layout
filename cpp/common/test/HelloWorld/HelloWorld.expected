| test.cpp:1:29:1:41 | Hello World |
| test.cpp:2:35:2:50 | Hello World!!! |
